astropy/io/fits/tests/test_core.py

```diff
@@ -593,15 +594,48 @@ def test_open_nonexistent(self):
             assert os.path.exists(self.temp('foobar.fits'))
             os.remove(self.temp('foobar.fits'))
 
+    def test_open_file_handle(self):
+        # Make sure we can open a FITS file from an open file handle
+        with open(self.data('test0.fits'), 'rb') as handle:
+            with fits.open(handle) as fitsfile:
+                pass
+
+        with open(self.temp('temp.fits'), 'wb') as handle:
+            with fits.open(handle) as fitsfile:
+                pass
+
+        # Opening without explicitly specifying binary mode should fail
```

Comment:
Binary mode is required, but it's tricky because it was the default on Python 2.